src/kaleidoscope/driver/keyscanner/AVR.h

```diff
@@ -0,0 +1,239 @@
+/* -*- mode: c++ -*-
+ * kaleidoscope::driver::keyscanner::AVR -- AVR-based keyscanner component
+ * Copyright (C) 2018-2019  Keyboard.io, Inc
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <Arduino.h>
+#include "Kaleidoscope-HIDAdaptor-KeyboardioHID.h"
+
+#include "kaleidoscope/macro_helpers.h"
+#include "kaleidoscope/driver/keyscanner/Base.h"
+#include "kaleidoscope/device/avr/pins_and_ports.h"
+
+#include <avr/wdt.h>
+
+#define ROW_PIN_LIST(...)  __VA_ARGS__
+#define COL_PIN_LIST(...)  __VA_ARGS__
+
+#define AVR_KEYSCANNER_PROPS(ROW_PINS_, COL_PINS_)               \
+  KEYSCANNER_PROPS(NUM_ARGS(ROW_PINS_), NUM_ARGS(COL_PINS_));       \
+  static constexpr uint8_t matrix_row_pins[matrix_rows] =  ROW_PINS_;   \
+  static constexpr uint8_t matrix_col_pins[matrix_columns] =  COL_PINS_;
+
+#define AVR_KEYSCANNER_BOILERPLATE()                                                      \
```

Comment:
Mostly because `AVR_KEYSCANNER_BOILERPLATE();` looks marginally better than `AVR_KEYSCANNER_BOILERPLATE;`. So aesthetic and hubris reasons only.

@obra, what would you prefer here?

Reply:
IMHO, there's no point in trying to make something that is not a function call look like it was one. 
Also, the semicolon at the end of the macro invocation isn't necessary anyway (null statement). Macro invocations should clearly be recognizable as such. Therefore, the both the `()` and the `;` are misleading.

Reply:
I don't think I feel strongly

Reply:
I went with `AVR_KEYSCANNER_BOILERPLATE`, no `()`, no `;`, just the macro.